Providers/Profile.swift

```diff
@@ -896,31 +971,42 @@ public class BrowserProfile: Profile {
         private func syncClientsWithDelegate(delegate: SyncDelegate, prefs: Prefs, ready: Ready) -> SyncResult {
             log.debug("Syncing clients to storage.")
             let clientSynchronizer = ready.synchronizer(ClientsSynchronizer.self, delegate: delegate, prefs: prefs)
-            return clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info)
+            let result = clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info)
+            syncDisplayState = displayStateForSyncResult(result)
```

Comment:
Mentioned in the comment above but I think by calling `.value` within `displayStateForSyncResult`, this method will block until the synchronizer is complete.